src/Core/Settings.h

```diff
@@ -560,6 +560,7 @@ class IColumn;
     M(Bool, asterisk_include_alias_columns, false, "Include ALIAS columns for wildcard query", 0) \
     M(Bool, optimize_skip_merged_partitions, false, "Skip partitions with one part with level > 0 in optimize final", 0) \
     M(Bool, optimize_on_insert, true, "Do the same transformation for inserted block of data as if merge was done on this block.", 0) \
+    M(Bool, optimize_use_projections, true, "Automatically choose projections to perform SELECT query", 0) \
```

Comment:
By the way it could be marked as an alias for `allow_experimental_projection_optimization` (https://github.com/ClickHouse/ClickHouse/pull/45659)

Reply:
@azat I like the idea, please do it.

Reply:
#50044

Also this will allow to remove `pr-backward-incompatible` label I guess